ka9q-radio fails to build with glibc < 2.34

libpthread was integrated into glibc in version 2.34 (released August 2021). As a result, it is no longer necessary to explicitly link to it with `-lpthread`.

In e410528ed3ea82d5aad3cf313f06a1d9e4a69936, `-lpthread` was removed, which means that ka9q-radio no longer builds with earlier glibc versions:

```
cc  -o aprs aprs.o libradio.a -lbsd -lm
/usr/bin/ld: libradio.a(misc.o): in function `realtime':
misc.c:(.text+0x249): undefined reference to `pthread_getname_np'
/usr/bin/ld: misc.c:(.text+0x298): undefined reference to `pthread_getname_np'
/usr/bin/ld: libradio.a(misc.o): in function `norealtime':
misc.c:(.text+0x3be): undefined reference to `pthread_getname_np'
/usr/bin/ld: misc.c:(.text+0x430): undefined reference to `pthread_getname_np'
collect2: error: ld returned 1 exit status
make: *** [Makefile.linux:122: aprs] Error 1
```

Thus older OSes like Debian 11 and Ubuntu 20.04 are no longer supported.

Was this change intentional? If not, then `-lpthread` should be added back for programs that need it. It may also be worth extending CI to build ka9q-radio on multiple OS versions to catch accidental breakage of this sort.

--- a/Makefile.linux
+++ b/Makefile.linux
@@ -119,81 +119,81 @@ dynamics: $(DYNAMIC_DRIVERS)
 # Executables
 
 aprs: aprs.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm -lpthread
 
 aprsfeed: aprsfeed.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm -lpthread
 
 control: control.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -liniparser -lncursesw -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -liniparser -lncursesw -lbsd -lm -lpthread
 
 cwd: cwd.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm -lpthread
 
 tune: tune.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm -lpthread
 
 setfilt: setfilt.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm -lpthread
 
 show-pkt: show-pkt.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lncursesw -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lncursesw -lbsd -lm -lpthread
 
 show-sig: show-sig.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lncursesw -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lncursesw -lbsd -lm -lpthread
 
 metadump: metadump.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm -lpthread
 
 monitor: monitor.o monitor-data.o monitor-display.o monitor-repeater.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lncursesw -lopus -lportaudio -lasound -lsamplerate -liniparser -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lncursesw -lopus -lportaudio -lasound -lsamplerate -liniparser -lbsd -lm -lpthread
 
 opusd: opusd.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lopus -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lopus -lbsd -lm -lpthread
 
 opussend: opussend.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lopus -lportaudio -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lopus -lportaudio -lbsd -lm -lpthread
 
 packetd: packetd.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lfftw3f_threads -lfftw3f -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lfftw3f_threads -lfftw3f -lbsd -lm -lpthread
 
 pcmcat: pcmcat.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm -lpthread
 
 pcmspawn: pcmspawn.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm -lpthread
 
 pcmrecord: pcmrecord.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lopus -logg -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lopus -logg -lbsd -lm -lpthread
 
 pcmsend: pcmsend.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lportaudio -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lportaudio -lbsd -lm -lpthread
 
 powers: powers.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm -lpthread
 
 pl: pl.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lfftw3f_threads -lfftw3f -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lfftw3f_threads -lfftw3f -lbsd -lm -lpthread
 
 ifeq ($(STATIC),1)
 radiod: main-static.o $(RADIOD_OBJECTS) $(STATIC_DRIVERS)
-	$(CC) -rdynamic $(LDOPTS) -o $@ $^  -lopus -lfftw3f_threads -lfftw3f -liniparser -lportaudio -lairspy -lairspyhf -lrtlsdr -lusb-1.0 -ldl -lbsd -lm
+	$(CC) -rdynamic $(LDOPTS) -o $@ $^  -lopus -lfftw3f_threads -lfftw3f -liniparser -lportaudio -lairspy -lairspyhf -lrtlsdr -lusb-1.0 -ldl -lbsd -lm -lpthread
 else
 radiod: main.o $(RADIOD_OBJECTS)
-	$(CC) -rdynamic $(LDOPTS) -o $@ $^  -lopus -lfftw3f_threads -lfftw3f -liniparser -ldl -lbsd -lm
+	$(CC) -rdynamic $(LDOPTS) -o $@ $^  -lopus -lfftw3f_threads -lfftw3f -liniparser -ldl -lbsd -lm -lpthread
 endif
 
 rdsd: rdsd.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lfftw3f_threads -lfftw3f -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lfftw3f_threads -lfftw3f -lbsd -lm -lpthread
 
 stereod: stereod.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lfftw3f_threads -lfftw3f -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lfftw3f_threads -lfftw3f -lbsd -lm -lpthread
 
 jt-decoded: jt-decoded.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm -lpthread
 
 wd-record: wd-record.o libradio.a
-	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm
+	$(CC) $(LDOPTS) -o $@ $^ -lbsd -lm -lpthread
 
 # not made unless we're on a rpi
 set_xcvr: set_xcvr.o libradio.a

